src/report.rs

```diff
@@ -3,6 +3,7 @@
 //! Build a report as clear as possible as to why
 //! dependency solving failed.
 
+use std::collections::HashSet;
```

Comment:
Nit: use `FxHashSet`?

Reply:
Hm it's not used consistently in the project